### PR TITLE
feat(rule_engine): default timeout for jq/2 and jq/3 with timeout

### PR DESCRIPTION
**Pull request by terry-xiaoyu**
This PR adds a default timeout of 10 seconds to the rule engine's jq/2 function, and adds a new function jq/3 (where the last parameter is a timeout value). The default timeout can be configured with the setting "rule_engine.jq_function_default_timeout".

Having a timeout when executing jq code in the rule engine is important as jq code can potentially run forever. Also, the Erlang jq library limits the number of jq programs that can execute concurrently so a jq program that loops forever could potentially also prevent a "non-buggy" jq program from ever starting.

I'm not sure that rule_engine.jq_function_default_timeout is the right place for the setting so please let me know if there is a better place. Also, I would like help to translate the documentation strings to Chinese.

This PR is just a proposal for how to add timeout to the jq function so we can still discuss alternative ways of implementing this.

### CODE REVIEW

**Comment by terry-xiaoyu** [comment]
This is a copy of https://github.com/emqx/emqx/pull/7975, to add my commits.

File: rebar.config.erl [hunk 1]
```diff
@@ -41,7 +41,7 @@ quicer() ->
     {quicer, {git, "https://github.com/emqx/quic.git", {tag, "0.0.9"}}}.
 
 jq() ->
-    {jq, {git, "https://github.com/emqx/jq", {tag, "v0.2.2"}}}.
+    {jq, {git, "https://github.com/emqx/jq", {tag, "v0.3.1"}}}.
```
Comment (zmstone) [inline rebar.config.erl:44]: I have pushed a new tag to fix a bug in ci which caused release package named without version number for certain platforms.
```suggestion
    {jq, {git, "https://github.com/emqx/jq", {tag, "v0.3.2"}}}.
```

File: mix.exs [hunk 2]
```diff
@@ -611,7 +611,7 @@ defmodule EMQXUmbrella.MixProject do
 
   defp jq_dep() do
     if enable_jq?(),
-      do: [{:jq, github: "emqx/jq", tag: "v0.2.2", override: true}],
+      do: [{:jq, github: "emqx/jq", tag: "v0.3.1", override: true}],
```
Comment (zmstone) [inline mix.exs:614]:
```suggestion
      do: [{:jq, github: "emqx/jq", tag: "v0.3.2", override: true}],
```

File: rebar.config.erl (hunk 1 above)
Comment (terry-xiaoyu) [reply rebar.config.erl:44]: Got it, I'll update the version number.

**Pull request merged**